File: LibroContableFrontEnd/src/app/core/services/movimiento.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { Movimiento, MovimientoRequest, MovimientoResponse } from '../models/movimiento.model';

@Injectable({
  providedIn: 'root'
})
export class MovimientoService {
  private readonly apiUrl = '/api/v1/movimientos';

  constructor(private readonly http: HttpClient) {}

  registrar(request: MovimientoRequest): Observable<MovimientoResponse> {
    return this.http.post<MovimientoResponse>(this.apiUrl, request);
  }

  actualizar(id: number, request: MovimientoRequest): Observable<MovimientoResponse> {
    return this.http.put<MovimientoResponse>(`${this.apiUrl}/${id}`, request);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  listarPorMes(mesId: number): Observable<Movimiento[]> {
    return this.http.get<Movimiento[]>(this.apiUrl, {
      params: { mesId: String(mesId) }
    });
  }
}

File: LibroContableBackEnd/src/main/java/com/mguevara/librocontable/Services/MovimientoService.java
package com.mguevara.librocontable.Services;

import java.util.List;

import com.mguevara.librocontable.Dto.Request.MovimientoRequest;
import com.mguevara.librocontable.Dto.Response.MovimientoResponse;
import com.mguevara.librocontable.Entity.Movimiento;

public interface MovimientoService {

    MovimientoResponse registrar(MovimientoRequest request);

    MovimientoResponse actualizar(Long id, MovimientoRequest request);

    void eliminar(Long id);

    List<Movimiento> listarPorMes(Long mesId);

}
